artiq/master/worker_impl.py

```diff
@@ -254,6 +254,9 @@ def put_exception_report():
         exc_str = str(exc)
         if exc_str:
             short_exc_info += ": " + exc_str.splitlines()[0]
+        for channel_num, replacer in device_mgr.lib.items():
+            if channel_num in short_exc_info:
+                short_exc_info = short_exc_info.replace(channel_num, replacer)
```

Comment:
As I had explained to you before, this is not the right way to do it. It is even buggy as it will indiscriminately replace numbers in user-defined exception strings.